libgadget/blackhole.c

```diff
@@ -1361,14 +1521,23 @@ blackhole_feedback_copy(int i, TreeWalkQueryBHFeedback * I, TreeWalk * tw)
     I->BH_Mass = BHP(i).Mass;
     I->ID = P[i].ID;
     I->Mtrack = BHP(i).Mtrack;
+    I->Density = BHP(i).Density;
     int PI = P[i].PI;
 
     I->FeedbackWeightSum = BH_GET_PRIV(tw)->BH_FeedbackWeightSum[PI];
+    I->FdbkChannel = 0; /* thermal feedback mode */
 
     double dtime = get_dloga_for_bin(P[i].TimeBin, P[i].Ti_drift) / BH_GET_PRIV(tw)->hubble;
 
     I->FeedbackEnergy = blackhole_params.BlackHoleFeedbackFactor * 0.1 * BHP(i).Mdot * dtime *
                 pow(LIGHTCGS / All.UnitVelocity_in_cm_per_s, 2);
+    I->KEFeedbackEnergy = 0;
+    if (blackhole_params.BlackHoleKineticOn == 1 && BH_GET_PRIV(tw)->KEflag[PI] > 0){
+        I->FdbkChannel = 1; /* kinetic feedback mode */
+        if (BH_GET_PRIV(tw)->KEflag[PI] == 2){
```

Comment:
I see - so the condition is really there to disable the thermal feedback, because thermal feedback cannot happen when KEfeedback energy is accumulating? Could you please add a comment.